### PR TITLE
Deepak implementation

### PR DESCRIPTION
**Pull request by Deepak-webelight**
- git local system setup and repo setup 
- Express Server setup for Ecommerce-demo 
- mongodb connection build
- folder structure setup
- UserModel added for authentication
- signup route added
- /auth/signup route middleware implemented - ( varifySignupRequestBody,varifyUserAlreadyExist  )

### CODE REVIEW

**Comment by Deepak-webelight** [comment]
- login route middleware implemented - ( varifyLoginRequestBody, varifyUserDoesNotExist )
- jsonwebtoken implemented

**Comment by Deepak-webelight** [comment]
- logout route and refresh token route implemented
- varify token middleware implemented

File: server/app.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import express, { Request, Response, Express } from "express";
+import cors from "cors";
+import authRoutes from "./routes/auth";
+
+const app: Express = express();
```
Comment (anshs-webelight) [inline server/app.ts]: No need to give explicit type here

File: server/.env [hunk 2]
```diff
@@ -0,0 +1,3 @@
+PORT = 5000
```
Comment (anshs-webelight) [inline server/.env]: We should include `.env` file to .gitignore

File: server/controller/auth/AddNewUserHandler/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import responseProvider from "../../../utils/responseProvider";
+import { IsignupRequestbody } from "../../../routes/auth";
+import UserModel from "../../../model/userModel";
+import tokenHandler from "../../../utils/tokenHandler";
+
+const AddNewUserHandler = async (req: Request, res: Response) => {
+  try {
+    const { Name, Email, Password }: IsignupRequestbody = req.body;
+    
+    const token = tokenHandler.generageToken({ Email, Name });
```
Comment (anshs-webelight) [inline server/controller/auth/AddNewUserHandler/index.ts]: we should create token after storing user to the database

File: server/model/userModel/index.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  Email: string;
+  Password: string;
+  access_token: string;
+}
+
+interface IUserModelSchema extends IUserModel, Document {}
+
+const userModelSchema = new Schema<IUserModelSchema>(
+  {
+    Name: {
```
Comment (anshs-webelight) [inline server/model/userModel/index.ts]: `Name` ==> `name`

File: server/model/userModel/index.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      type: String,
+      required: true,
+    },
+    Email: {
```
Comment (anshs-webelight) [inline server/model/userModel/index.ts]: same naming suggestion as name

File: server/utils/envProvider.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { config } from "dotenv";
+config();
+
+interface IenvProvider {
+  PORT: string | undefined;
+  MONGODB_URL: string | undefined;
+  TOKEN_SECRET: string | undefined;
+}
+
+const envProvider: IenvProvider = {
```
Comment (anshs-webelight) [inline server/utils/envProvider.ts]:
```suggestion
const envProvider = {
```

File: server/utils/envProvider.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  PORT: process.env.PORT,
+  MONGODB_URL: process.env.MONGODB_URL,
+  TOKEN_SECRET: process.env.TOKEN_SECRET,
```
Comment (anshs-webelight) [inline server/utils/envProvider.ts]:
```suggestion
  port: process.env.PORT,
  mongodbUrl: process.env.MONGODB_URL,
  jwtSecret: process.env.TOKEN_SECRET,
```

File: server/utils/tokenHandler.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import jwt from "jsonwebtoken";
+import envProvider from "./envProvider";
+
+export interface IgenerageTokenInput {
+  Name: string;
+  Email: string;
+}
+
+interface ItokenHandler {
+  generageToken: ({}: IgenerageTokenInput) => string;
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.ts]:
```suggestion
  generageToken: (data: IgenerageTokenInput) => string;
```

File: server/utils/tokenHandler.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  verifyToken: (token: string) => jwt.JwtPayload | string;
+}
+
+const tokenHandler: ItokenHandler = {
+  generageToken: ({ Name, Email }) => {
+    if (!envProvider.TOKEN_SECRET) throw new Error("Invalid token secret");
+    const token = jwt.sign({ Name, Email }, envProvider.TOKEN_SECRET, {
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.ts]: no need to add name and email in token, we should add only user id

File: server/utils/tokenHandler.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+      expiresIn: "24h",
+    });
+    return token;
+  },
+  verifyToken: async (token) => {
+    if (!envProvider.TOKEN_SECRET) throw new Error("Invalid token secret");
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.ts]:
```suggestion
    if (!envProvider.TOKEN_SECRET) throw Error("Invalid token secret");
```

File: server/package.json [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "author": "Deepak Mandal",
+  "license": "ISC",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "dotenv": "^16.4.5",
+    "express": "^4.19.2",
+    "jsonwebtoken": "^9.0.2",
+    "mongoose": "^8.3.1",
+    "nodemon": "^3.1.0"
```
Comment (anshs-webelight) [inline server/package.json]: nodemon should be installed as devDependencies

**Review by anshs-webelight — CHANGES_REQUESTED** [review]

**Comment by Deepak-webelight** [comment]
- class implemented in controllers middleware and services
- removed env from github
- file and folder structure updated
- sign-up endpoint completed

File: server/config/connectDB.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import mongoose from "mongoose";
+import envProvider from "../utils/envProvider.utils";
```
Comment (anshs-webelight) [inline server/config/connectDB.ts]: moved this file to config folder and change name `envProvider` to `appConfig`

File: server/middleware/user.middleware.ts [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+          statusCode: 400,
+        });
+      }
+      const isValidToken = await tokenHandler.verifyToken(token);
+
+      // validate active user with token
+
+      const validateActiveUser = await userService.findUserData({
+        access_token: token,
+      });
```
Comment (anshs-webelight) [inline server/middleware/user.middleware.ts]: no need to find active user from database we can get userId from token only

Comment (anshs-webelight) [reply server/middleware/user.middleware.ts]: And why are we storing access-token in the database? there is no need of that

File: server/models/user.model.ts [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+  password: string;
+  access_token: string;
+}
+
+export interface IuserDocument extends IuserModel, Document {}
+
+const userModelSchema = new Schema<IuserDocument>(
+  {
+    name: {
+      type: String,
```
Comment (anshs-webelight) [inline server/models/user.model.ts:14]: you can use `unique: true` as well

File: server/models/user.model.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      required: true,
+    },
+    email: {
+      type: String,
```
Comment (anshs-webelight) [inline server/models/user.model.ts:18]: add validation function which validates email, is it in correct format or not

File: server/routes/user.routes.ts [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+// user logout route
+userRoutes.get(
+  "/logout",
+  usermiddlware.varifyAuthToken,
+  userController.userLogout
+);
+
+// refresh token route
+userRoutes.get(
```
Comment (anshs-webelight) [inline server/routes/user.routes.ts]:
```suggestion
userRoutes.post(
```

File: server/routes/user.routes.ts [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+  password: string;
+}
+userRoutes.post(
+  "/login",
+  usermiddlware.varifyLoginRequestBody,
+  userController.userLogin
+);
+
+// user logout route
+userRoutes.get(
```
Comment (anshs-webelight) [inline server/routes/user.routes.ts]:
```suggestion
userRoutes.post(
```

File: server/services/user.service.ts [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+      // Check if user already exists
+      const userExist = await this.isUserExist(email);
+      if (userExist) {
+        return Error("User already exists");
+      }
+      // register as new user
+      const registeredUser = await new userModel({
+        email,
+        name,
+        password,
```
Comment (anshs-webelight) [inline server/services/user.service.ts]: we should not store password in database as it is, need to store hash passwords

Comment (anshs-webelight) [reply server/services/user.service.ts]: check `bcryptjs` package for this

File: server/services/user.service.ts [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+    access_token,
+  }: IResisterRequestbody): Promise<IuserDocument | Error> {
+    try {
+      // Check if user already exists
+      const userExist = await this.isUserExist(email);
+      if (userExist) {
+        return Error("User already exists");
+      }
+      // register as new user
+      const registeredUser = await new userModel({
```
Comment (anshs-webelight) [inline server/services/user.service.ts]: no need to use await here

File: server/services/user.service.ts [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+        email,
+        name,
+        password,
+        access_token,
+      });
+      const savedUser = registeredUser.save();
+
+      return savedUser;
```
Comment (anshs-webelight) [inline server/services/user.service.ts]:
```suggestion
      return registeredUser.save()
```

File: server/services/user.service.ts [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+  }
+  async validiateUser({
+    email,
+    password,
+  }: ILoginRequestbody): Promise<IuserDocument | Error> {
+    try {
+      // if user not existing
+      const isUserExist = await this.isUserExist(email);
+      if (!isUserExist) {
+        return Error("User does not exist");
```
Comment (anshs-webelight) [inline server/services/user.service.ts]: why we are returning Error?

Comment (anshs-webelight) [reply server/services/user.service.ts]:
```suggestion
        throw Error("User does not exist");
```

File: server/services/user.service.ts [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+      if (response) return response;
+      else return Error("user update failed");
+    } catch (err) {
+      throw new Error((err as Error).message);
+    }
+  }
+  async findUserData(
+    searchQuery: IUserSearchQuery
+  ): Promise<IuserDocument | Error> {
+    try {
```
Comment (anshs-webelight) [inline server/services/user.service.ts:79]: service file should not contain try-catch block, error will be handled at the controller (parent) side catch block

File: server/services/user.service.ts [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+      const response = await userModel.findOne(searchQuery);
+      if (response) return response;
+      else return Error("user not found");
```
Comment (anshs-webelight) [inline server/services/user.service.ts]:
```suggestion
      if(!response) {
          throw Error("user not found");
      }
      
      return response
```

File: server/services/user.service.ts [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+      else return Error("Invalid email or password");
+    } catch (err) {
+      throw new Error((err as Error).message);
+    }
+  }
+  async updateUserData(
+    searchQuery: IUserSearchQuery,
+    filterQuery: IUserFilterQuery
+  ): Promise<IuserDocument | Error> {
+    try {
```
Comment (anshs-webelight) [inline server/services/user.service.ts]: remove try-catch from service file

File: server/utils/tokenHandler.utils.ts [hunk 25]
```diff
@@ -0,0 +1,30 @@
+import jwt from "jsonwebtoken";
+import envProvider from "./envProvider.utils";
+
+export interface IgenerageTokenInput {
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.utils.ts]:
```suggestion
export interface IgenerateToken {
```

File: server/utils/tokenHandler.utils.ts [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+  userId: string;
+}
+
+interface ItokenHandler {
+  generageToken: (data: IgenerageTokenInput) => string;
+  verifyToken: (token: string) => jwt.JwtPayload | string;
+}
+
+const tokenHandler: ItokenHandler = {
+  generageToken: ({ userId }) => {
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.utils.ts]:
```suggestion
  generateToken: ({ userId }) => {
```

File: server/utils/tokenHandler.utils.ts [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+    if (!envProvider.jwtSecret) throw Error("Invalid token secret");
+    const token = jwt.sign({ userId }, envProvider.jwtSecret, {
+      expiresIn: "24h",
+    });
+    return token;
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.utils.ts]:
```suggestion
    return jwt.sign({ userId }, envProvider.jwtSecret, {
      expiresIn: "24h",
    });
```

File: server/utils/tokenHandler.utils.ts [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  verifyToken: async (token) => {
+    if (!envProvider.jwtSecret) throw Error("Invalid token secret");
+    try {
+      return await jwt.verify(token, envProvider.jwtSecret);
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.utils.ts]: why do we use await here? there is no need for that

File: server/utils/tokenHandler.utils.ts [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+
+const tokenHandler: ItokenHandler = {
+  generageToken: ({ userId }) => {
+    if (!envProvider.jwtSecret) throw Error("Invalid token secret");
+    const token = jwt.sign({ userId }, envProvider.jwtSecret, {
+      expiresIn: "24h",
+    });
+    return token;
+  },
+  verifyToken: async (token) => {
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.utils.ts]:
```suggestion
  verifyToken: (token) => {
```

File: server/controllers/user.controller.ts [hunk 30]
```diff
@@ -0,0 +1,153 @@
...
+      });
+    }
+  }
+  async refreshToken(req: Request, res: Response): Promise<void> {
+    try {
+      const { authorization } = req.headers;
+
+      const token = (authorization as string)?.split(" ")[1];
+
+      const tokenData = (await tokenHandler.verifyToken(
```
Comment (anshs-webelight) [inline server/controllers/user.controller.ts]:
```suggestion
      const tokenData = (tokenHandler.verifyToken(
```

File: server/controllers/user.controller.ts [hunk 31]
```diff
@@ -0,0 +1,153 @@
...
+        token
+      )) as IgenerageTokenInput;
+
+      const newToken = await tokenHandler.generageToken({
```
Comment (anshs-webelight) [inline server/controllers/user.controller.ts]:
```suggestion
      const newToken =  tokenHandler.generageToken({
```

File: server/utils/tokenHandler.utils.ts [hunk 32]
```diff
@@ -0,0 +1,30 @@
...
+export interface IgenerageTokenInput {
+  userId: string;
+}
+
+interface ItokenHandler {
+  generageToken: (data: IgenerageTokenInput) => string;
+  verifyToken: (token: string) => jwt.JwtPayload | string;
+}
+
+const tokenHandler: ItokenHandler = {
```
Comment (anshs-webelight) [inline server/utils/tokenHandler.utils.ts]: no need to create a tokenHandler object you can define both methods separately and export it from this file

**Review by anshs-webelight — CHANGES_REQUESTED** [review]

File: server/config/appConfig.ts [hunk 33]
```diff
@@ -0,0 +1,16 @@
+import { config } from "dotenv";
+config();
+
+interface IappConfig {
+  port: string | undefined;
+  mongodbUrl: string | undefined;
+  jwtSecret: string | undefined;
+}
+
+const appConfig: IappConfig = {
```
Comment (anshs-webelight) [inline server/config/appConfig.ts:10]: no need to give explicit type

File: server/config/connectDB.ts [hunk 34]
```diff
@@ -0,0 +1,15 @@
+import mongoose from "mongoose";
+import appConfig from "./appConfig";
+
+const connectDb = async () => {
+  if (!appConfig.mongodbUrl) throw new Error("MONGODB_URI is required");
```
Comment (anshs-webelight) [inline server/config/connectDB.ts:5]:
```suggestion
  if (!appConfig.mongodbUrl) throw Error("MONGODB_URI is required");
```

File: server/controllers/user.controller.ts [hunk 35]
```diff
@@ -0,0 +1,174 @@
...
+        statusCode: 200,
+        cookie: [
+          {
+            name: "refreshToken",
+            value: tokenFormat(refreshToken),
+            cookieOptions: { httpOnly: true, sameSite: "strict" },
+          },
+          {
+            name: "token",
+            value: tokenFormat(refreshToken),
```
Comment (anshs-webelight) [inline server/controllers/user.controller.ts]:
```suggestion
            name: "token",
            value: tokenFormat(token),
```

File: server/controllers/user.controller.ts [hunk 36]
```diff
@@ -0,0 +1,174 @@
...
+
+      return responseProvider.sendResponse({
+        message: "User Created Successfully",
+        response: res,
+        statusCode: 200,
+        cookie: [
+          {
+            name: "refreshToken",
+            value: tokenFormat(refreshToken),
+            cookieOptions: { httpOnly: true, sameSite: "strict" },
```
Comment (anshs-webelight) [inline server/controllers/user.controller.ts]: add this  cookieOptions at responseProvider utils sendResponse method no need to pass it from controller

File: server/models/user.model.ts [hunk 37]
```diff
@@ -0,0 +1,31 @@
...
+      required: true,
+      unique: true,
+    },
+    password: {
+      type: String,
+      required: true,
+    },
+  },
+  { versionKey: false }
```
Comment (anshs-webelight) [inline server/models/user.model.ts:28]:
```suggestion
  { versionKey: false,  timestamps: true }
```

File: server/config/connectDB.ts (hunk 12 above)
Comment (Deepak-webelight) [reply server/config/connectDB.ts]: done

File: server/middleware/user.middleware.ts (hunk 13 above)
Comment (Deepak-webelight) [reply server/middleware/user.middleware.ts]: done both

File: server/models/user.model.ts (hunk 14 above)
Comment (Deepak-webelight) [reply server/models/user.model.ts:14]: implemented email as unique : true

File: server/models/user.model.ts (hunk 15 above)
Comment (Deepak-webelight) [reply server/models/user.model.ts:18]: this is implemented in middleware payload is verifying for /sign-up and /login end point

File: server/services/user.service.ts (hunk 19 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts]: updated

File: server/routes/user.routes.ts (hunk 16 above)
Comment (Deepak-webelight) [reply server/routes/user.routes.ts]: updated

File: server/routes/user.routes.ts (hunk 17 above)
Comment (Deepak-webelight) [reply server/routes/user.routes.ts]: updated

File: server/services/user.service.ts (hunk 18 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts]: implemented as hashpassword service and verifypassword service

File: server/services/user.service.ts (hunk 20 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts]: updated

File: server/services/user.service.ts (hunk 21 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts]: updated

File: server/services/user.service.ts (hunk 22 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts:79]: updated

File: server/services/user.service.ts (hunk 23 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts]: updated

File: server/services/user.service.ts (hunk 24 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts]: updated

File: server/utils/tokenHandler.utils.ts (hunk 25 above)
Comment (Deepak-webelight) [reply server/utils/tokenHandler.utils.ts]: updated

File: server/utils/tokenHandler.utils.ts (hunk 26 above)
Comment (Deepak-webelight) [reply server/utils/tokenHandler.utils.ts]: updated

File: server/utils/tokenHandler.utils.ts (hunk 27 above)
Comment (Deepak-webelight) [reply server/utils/tokenHandler.utils.ts]: updated

File: server/utils/tokenHandler.utils.ts (hunk 28 above)
Comment (Deepak-webelight) [reply server/utils/tokenHandler.utils.ts]: updated

File: server/utils/tokenHandler.utils.ts (hunk 29 above)
Comment (Deepak-webelight) [reply server/utils/tokenHandler.utils.ts]: updated

File: server/controllers/user.controller.ts (hunk 30 above)
Comment (Deepak-webelight) [reply server/controllers/user.controller.ts]: updated

File: server/controllers/user.controller.ts (hunk 31 above)
Comment (Deepak-webelight) [reply server/controllers/user.controller.ts]: updated

File: server/utils/tokenHandler.utils.ts (hunk 32 above)
Comment (Deepak-webelight) [reply server/utils/tokenHandler.utils.ts]: both methods is now as a  service

**Review by anshs-webelight — CHANGES_REQUESTED** [review]

File: server/utils/responseProvider.utils.ts [hunk 38]
```diff
@@ -0,0 +1,95 @@
+import { Response } from "express";
+
+// user route Response data
+
+type ServerResponseData = any; // todo
+
+interface ICookieOptions {
+  domain?: string;
+  path?: string;
+  expires?: Date | undefined;
```
Comment (anshs-webelight) [inline server/utils/responseProvider.utils.ts:10]:
```suggestion
  expires?: Date;
```

File: server/utils/responseProvider.utils.ts [hunk 39]
```diff
@@ -0,0 +1,95 @@
+import { Response } from "express";
+
+// user route Response data
+
+type ServerResponseData = any; // todo
```
Comment (anshs-webelight) [inline server/utils/responseProvider.utils.ts:5]: don't use any as type

File: server/services/user.service.ts [hunk 40]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  createNewToken(userId: string, expiresIn: "1d" | "3m"): string {
+    if (!appConfig.jwtSecret) {
+      throw Error("Invalid token secret");
+    }
+    return jwt.sign({ userId }, appConfig.jwtSecret, {
+      expiresIn,
+    });
+  }
+  verifyToken(token: string): jwt.JwtPayload | string | boolean {
```
Comment (anshs-webelight) [inline server/services/user.service.ts:77]: move this two functions into `/utils/jwt.ts` file

File: server/services/user.service.ts [hunk 41]
```diff
@@ -0,0 +1,102 @@
...
+      const hashedPassword = await bcrypt.hash(password, salt);
+      return hashedPassword;
+    } catch (error) {
+      throw error;
+    }
+  }
+  async verifyPassword(plainPassword: string, hashedPassword: string) {
+    try {
+      const isMatch = await bcrypt.compare(plainPassword, hashedPassword);
+      return isMatch;
```
Comment (anshs-webelight) [inline server/services/user.service.ts:97]:
```suggestion
     return bcrypt.compare(plainPassword, hashedPassword);
```

File: server/services/user.service.ts [hunk 42]
```diff
@@ -0,0 +1,102 @@
+import appConfig from "../config/appConfig";
+import userModel, { IuserDocument } from "../models/user.model";
+import { ILoginRequestbody, IResisterRequestbody } from "../routes/user.routes";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcryptjs";
```
Comment (anshs-webelight) [inline server/services/user.service.ts:5]:
```suggestion
import { sign, JwtPayload, verify } from "jsonwebtoken";
import { hash, compare, genSalt } from "bcryptjs";
```

**Review by anshs-webelight — CHANGES_REQUESTED** [review]

File: server/config/appConfig.ts (hunk 33 above)
Comment (Deepak-webelight) [reply server/config/appConfig.ts:10]: done

File: server/config/connectDB.ts (hunk 34 above)
Comment (Deepak-webelight) [reply server/config/connectDB.ts:5]: done

File: server/controllers/user.controller.ts (hunk 35 above)
Comment (Deepak-webelight) [reply server/controllers/user.controller.ts]: fixed

File: server/controllers/user.controller.ts (hunk 36 above)
Comment (Deepak-webelight) [reply server/controllers/user.controller.ts]: fixed

File: server/models/user.model.ts (hunk 37 above)
Comment (Deepak-webelight) [reply server/models/user.model.ts:28]: added

File: server/utils/responseProvider.utils.ts (hunk 38 above)
Comment (Deepak-webelight) [reply server/utils/responseProvider.utils.ts:10]: fixed

File: server/utils/responseProvider.utils.ts (hunk 39 above)
Comment (Deepak-webelight) [reply server/utils/responseProvider.utils.ts:5]: fixed

File: server/services/user.service.ts (hunk 40 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts:77]: updated

File: server/services/user.service.ts (hunk 41 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts:97]: updated

File: server/services/user.service.ts (hunk 42 above)
Comment (Deepak-webelight) [reply server/services/user.service.ts:5]: fixed

**Pull request merged**